cmd/ncproxy/hcn_networking_test.go

```diff
@@ -0,0 +1,1200 @@
+		t.Fatalf("failed to create a test ncproxy networking store with %v", err)
+	}
+	defer closer()
+
+	// setup test ncproxy grpc service
+	agentCache := newComputeAgentCache()
+	gService := newGRPCService(agentCache, networkingStore)
+
+	// test network
+	network, err := createTestNATNetwork(t.Name() + "network")
+	if err != nil {
+		t.Fatalf("failed to create test network with %v", err)
+	}
+	defer func() {
+		_ = network.Delete()
+	}()
+
+	endpoint, err := createTestEndpoint(testEndpointName, network.Id)
+	if err != nil {
+		t.Fatalf("failed to create test endpoint with %v", err)
+	}
+	// defer cleanup in case of error. ignore error from the delete call here
+	// since we may have already successfully deleted the endpoint.
+	defer func() {
+		_ = endpoint.Delete()
+	}()
+
+	// create mocked compute agent service
+	computeAgentCtrl := gomock.NewController(t)
+	defer computeAgentCtrl.Finish()
+	mockedService := computeagentMock.NewMockComputeAgentService(computeAgentCtrl)
+	mockedAgentClient := &computeAgentClient{nil, mockedService}
+
+	// put mocked compute agent in agent cache for test
+	if err := agentCache.put(containerID, mockedAgentClient); err != nil {
+		t.Fatal(err)
+	}
+
+	// setup expected mocked calls
+	mockedService.EXPECT().DeleteNIC(gomock.Any(), gomock.Any()).Return(&computeagent.DeleteNICInternalResponse{}, nil).AnyTimes()
+
+	type config struct {
+		name          string
+		containerID   string
+		nicID         string
+		endpointName  string
+		errorExpected bool
+	}
+	tests := []config{
+		{
+			name:          "DeleteNIC returns no error",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: false,
+		},
+		{
+			name:          "DeleteNIC returns error with blank container ID",
+			containerID:   "",
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "DeleteNIC returns error with blank nic ID",
+			containerID:   containerID,
+			nicID:         "",
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "DeleteNIC returns error with blank endpoint name",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  "",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(_ *testing.T) {
+			req := &ncproxygrpc.DeleteNICRequest{
+				ContainerID:  test.containerID,
+				NicID:        test.nicID,
+				EndpointName: test.endpointName,
+			}
+
+			_, err := gService.DeleteNIC(ctx, req)
+			if test.errorExpected && err == nil {
+				t.Fatalf("expected DeleteNIC to return an error")
+			}
+			if !test.errorExpected && err != nil {
+				t.Fatalf("expected DeleteNIC to return no error, instead got %v", err)
+			}
+		})
+	}
+}
+
+func TestModifyNIC_HCN(t *testing.T) {
+	// support for setting IOV policy was added in 21H1
+	if osversion.Build() < osversion.V21H1 {
+		t.Skip("Requires build +21H1")
+	}
```

Comment:
Believe we'll need ncproxy/this directory to have a manifest file for osversion to function

Reply:
hmm, it seems to have been working already though. Do you know why that might be?

Reply:
`osversion.Build()` should return the Windows 8 OS version value if the application is not manifested, which is less than 21H1's, so it worked just by us getting lucky it seems haha. https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversion

```go
package main

import (
	"fmt"

	"github.com/Microsoft/hcsshim/osversion"
)

func main() {
	fmt.Println(osversion.Build(), osversion.V21H1)
}
```

should print "9200 19043". Unless by working you mean it wasn't skipping this test, then I'm not sure 😐